src/types/function.rs

```diff
@@ -17,56 +14,32 @@ impl PyCFunction {
     /// See [raw_pycfunction] for documentation on how to get the `fun` argument.
     pub fn new_with_keywords<'a>(
         fun: ffi::PyCFunctionWithKeywords,
-        name: &str,
+        name: &'static str,
         doc: &'static str,
         py_or_module: PyFunctionArguments<'a>,
     ) -> PyResult<&'a PyCFunction> {
-        let fun = PyMethodType::PyCFunctionWithKeywords(fun);
-        Self::new_(fun, name, doc, py_or_module)
+        Self::new_(
+            PyMethodDef::cfunction_with_keywords(name, fun, 0, doc),
```

Comment:
This removes error-handling of non-null terminated names and docs and instead causes panics in this API without documenting it. If panics are the way you want to handle non-terminated names and docs, it should at least be documented.